lib/chef/resource/windows_certificate.rb

```diff
@@ -162,24 +162,46 @@ class WindowsCertificate < Chef::Resource
 
       action_class do
         def add_cert(cert_obj)
-          store = ::Win32::Certstore.open(new_resource.store_name)
+          # store = ::Win32::Certstore.open()
+          store = ::Win32::Certstore.open(new_resource.store_name, get_cert_location)
           store.add(cert_obj)
         end
 
-        def add_pfx_cert
+        def add_pfx_cert(location)
           exportable = new_resource.exportable ? 1 : 0
-          store = ::Win32::Certstore.open(new_resource.store_name)
+          store = ::Win32::Certstore.open(new_resource.store_name, get_cert_location)
           store.add_pfx(new_resource.source, new_resource.pfx_password, exportable)
         end
 
         def delete_cert
-          store = ::Win32::Certstore.open(new_resource.store_name)
-          store.delete(new_resource.source)
+          store = ::Win32::Certstore.open(new_resource.store_name, get_cert_location)
+          if check_string(new_resource.source) && new_resource.source.length == 40
+            store.delete(new_resource.source)
+          else
+            thumbprint = powershell_out!(get_thumbprint(new_resource.store_name, get_cert_location, new_resource.source)).stdout
```

Comment:
this should use powershell_exec, not powershell_out, right @tas50 ?

Reply:
Yep. Powershell_exec should always be used

Reply:
Fixed